// Copyright (c) 2023 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;
option go_package = "bosdyn/api/image_service";

option java_outer_classname = "ImageServiceProto";

import "bosdyn/api/image.proto";

// An Image service provides access to one or more images, for example from cameras. It
// supports querying for the list of available images provided by the service and then
// supports requesting a latest given image by source name.
service ImageService {
    // Obtain the list of ImageSources for this given service.
    // Note that there may be multiple ImageServices running, each with their own set of sources
    // The name field keys access to individual images when calling GetImage.
    rpc ListImageSources(ListImageSourcesRequest) returns (ListImageSourcesResponse) {}

    // Request an image by name, with optional parameters for requesting image quality level.
    rpc GetImage(GetImageRequest) returns (GetImageResponse) {}

    // Request an image stream to live feed images.
    rpc GetImageStream(GetImageStreamRequest) returns (stream GetImageStreamResponse) {}

}
